src/library/dao/trackdao.h

```diff
@@ -73,21 +73,26 @@ class TrackDAO : public QObject, public virtual DAO, public virtual GlobalTrackC
     void saveTrack(Track* pTrack) const;
 
   signals:
+    // Forwarded from Track object
     void trackDirty(TrackId trackId) const;
     void trackClean(TrackId trackId) const;
-    void trackChanged(TrackId trackId);
-    void tracksChanged(QSet<TrackId> trackIds);
-    void tracksAdded(QSet<TrackId> trackIds);
-    void tracksRemoved(QSet<TrackId> trackIds);
-    void dbTrackAdded(TrackPointer pTrack);
+
+    // Multiple tracks
+    void tracksAdded(QSet<TrackId> trackIds) const;
+    void tracksChanged(QSet<TrackId> trackIds) const;
+    void tracksRemoved(QSet<TrackId> trackIds) const;
```

Comment:
Clazy complains about this because signals should not be `const`: https://github.com/KDE/clazy/blob/master/docs/checks/README-const-signal-or-slot.md

If we don't make it const, we have to make all methods that emit it non-const either. Opinions?

Reply:
The root cause is the inappropriate design. The database/storage components should not emit any signals and should strictly apply a request/response pattern, i.e. simple function calls with parameters and a return value.

Can we disable this warning programmatically for the DAO classes? I will try a workaround that allows us to keep the public methods const as intended.